include/mapbox/cheap_ruler.hpp

```diff
@@ -225,10 +228,7 @@ class CheapRuler {
                 }
             }
 
-            dx = (p.x - x) * kx;
-            dy = (p.y - y) * ky;
-
-            auto sqDist = dx * dx + dy * dy;
+            auto sqDist = squareDistance(p, {x, y});
```

Comment:
Would this be an added overhead because we're creating a new point object here? Not sure

Reply:
Zero overhead here ..